pkg/util/test/util.go

```diff
@@ -643,6 +644,71 @@ func validateImageTag(tag, namespace string, listOptions map[string]string) erro
 	return nil
 }
 
+func validateMonitoring(cluster *corev1alpha1.StorageCluster, timeout, interval time.Duration) error {
+	if cluster.Spec.Monitoring != nil &&
+		(*cluster.Spec.Monitoring.EnableMetrics ||
+			(cluster.Spec.Monitoring.Prometheus != nil && cluster.Spec.Monitoring.Prometheus.ExportMetrics)) {
+		if cluster.Spec.Monitoring.Prometheus != nil && cluster.Spec.Monitoring.Prometheus.Enabled {
+			dep := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "prometheus-operator",
+					Namespace: cluster.Namespace,
+				},
+			}
+			if err := appops.Instance().ValidateDeployment(&dep, timeout, interval); err != nil {
+				return err
+			}
+
+			st := appsv1.StatefulSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "prometheus-prometheus",
+					Namespace: cluster.Namespace,
+				},
+			}
+			if err := appops.Instance().ValidateStatefulSet(&st, timeout); err != nil {
+				return err
+			}
+		}
+		t := func() (interface{}, bool, error) {
+			alertManager, err := prometheusops.Instance().GetAlertManager("portworx", cluster.Namespace)
+			if err != nil {
+				return nil, true, err
+			}
+			if *alertManager.Spec.Replicas != alertManager.Status.AvailableReplicas {
+				return nil, true, fmt.Errorf("not all replicas are ready. expected: %d available: %d",
+					*alertManager.Spec.Replicas, alertManager.Status.AvailableReplicas)
+			}
+			return nil, false, nil
+		}
+		if _, err := task.DoRetryWithTimeout(t, timeout, interval); err != nil {
+			return err
+		}
+
+		t = func() (interface{}, bool, error) {
+			_, err := prometheusops.Instance().GetPrometheusRule("portworx", cluster.Namespace)
```

Comment:
nit: Can use constants from here for PrometheusRule and ServiceMonitor - 
https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/component/monitoring.go#L29